### PR TITLE
Replace Dep with Go Modules 

### LINKED ISSUE #171: Support Go Modules
##### SUMMARY

Ref #170  we will be moving from Go 1.10,1.11 to Go 1.11,1.12 and `go mod` is supported 1.11 onwards. So we should add support for this

##### ENVIRONMENT
* Kubeaudit version: 0.4.0
* Kubeaudit install method: -

- [x] If the maintainers agree with the feature as described here, I intend to submit a Pull Request myself.<sup>1</sup>

<sup><small>1</small></sup> <sub>This is the quickest way to get a new feature! We reserve the right to close feature requests, even ones we like, if the proposer does not intend to contribute to the feature and it doesn't fit in our current roadmap.</sub>

### PR DESCRIPTION
**Pull request by nschhina**
##### Description
Ref #171 we will be moving from Go 1.10,1.11 to Go 1.11,1.12 and `go mod` is supported 1.11 onwards. This PR adds the support for `go mod`

Fixes # (issue)
closes #171 
##### Type of change

- [x] New feature :sparkles:
- [x] Breaking changes :warning:

##### Checklist:

- [x] I have :tophat: my changes (A 🎩 specifically includes pulling down changes, setting them up, and manually testing the changed features and potential side effects to make sure nothing is broken)
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage did not decrease

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,8 +45,7 @@ check_version:
 	./$(BINARY_NAME) version
 
 setup:
-	$(GOCMD) get -u github.com/golang/dep/cmd/dep
-	$(GODEP) ensure
+	go mod download
```
Comment (klautcomputing) [inline Makefile]: I think we should add a `go mod tidy` somewhere

File: .gitignore [hunk 2]
```diff
@@ -4,7 +4,6 @@
 /.glide
 /.dev
 .DS_Store
-/vendor
```
Comment (klautcomputing) [inline .gitignore]: since you don't commit any thing in vendor we don't need to include it, do we?

Comment (nschhina) [reply .gitignore]: Correct, vendor is useless now, we'll not have any vendor files anymore. That's why I excluded it from gitignore

File: Makefile (hunk 1 above)
Comment (nschhina) [reply Makefile]: Done!

File: .gitignore (hunk 2 above)
Comment (klautcomputing) [reply .gitignore]: Okay this just means we all need to delete `vendor` going forward 👍

Comment (nschhina) [reply .gitignore]: We also need to `export GO111MODULE=on` if we are working on kubeaudit in $GOPATH

**Pull request merged**